.travis.yml

```diff
@@ -2,12 +2,27 @@ language: c
 sudo: required
 env:
   - TARGET=local
-  - TARGET=docker
+  - TARGET=docker OS=ubuntu-16.04
+  - TARGET=docker OS=fedora-25
 services:
   - docker
 before_install:
-  - if [ ${TARGET}="local" ];  then sudo apt-get update && sudo apt-get install libsqlite3-dev; fi
-  - if [ ${TARGET}="docker" ]; then sudo docker build -t "cangjians/libcangjie:latest" . ; fi
+  - | # before_install: local
+    if [ "${TARGET}" = "local"  ]; then
+      sudo apt-get update && sudo apt-get install libsqlite3-dev
+    fi
+  - | # before_install: docker
+    if [ "${TARGET}" = "docker" ]; then
+      sudo docker pull "cangjians/build-essential:${OS}"
+      sudo docker tag "cangjians/build-essential:${OS}" "cangjians/build-essential:latest"
+      sudo docker build -t "cangjians/libcangjie" .
```

Comment:
Sure. That doesn't change the fact that we could pre-build our own images and upload them to the Docker Hub, so that when running the CI we can just consume them, not build them, and thus make the CI that much faster?